streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java

```diff
@@ -151,7 +153,14 @@ public KStreamBuilder() {
         addProcessor(name, processorSupplier, source);
 
         final KTableImpl kTable = new KTableImpl<>(this, name, processorSupplier, Collections.singleton(source), keySerde, valSerde, storeName);
-        kTable.materialize((KTableSource) processorSupplier);
+        StateStoreSupplier storeSupplier = new RocksDBKeyValueStoreSupplier<>(storeName,
```

Comment:
Actually thinking about this a bit more, why do we always want to materialize the KTable source? If users do not provide a state store name when creating a KTable, it means they are not interested in querying it, so in this case we should probably not materialize it, no?